track.go

```diff
@@ -66,13 +72,29 @@ func newTrack(codecs []*webrtc.RTPCodec, trackGenerator TrackGenerator, d driver
 }
 
 func (t *track) OnEnded(handler func(error)) {
-	t.onErrorHandler.Store(handler)
+	t.mu.Lock()
+	t.onErrorHandler = handler
+	err := t.err
+	t.mu.Unlock()
```

Comment:
probably use `defer` instead?

Reply:
I want to Unlock before calling `handler`, since user can call `OnEnded` in the handler which causes deadlock if not unlocked.

Reply:
Ah, I see. You really thought ahead.... Love it.